src/validators/claims-validator.ts

```diff
@@ -27,3 +28,21 @@ export const areClaimsValid = (claims: string[], config: Config): boolean => {
 
   return true;
 };
+
+export const checkForReturnCode = (
+  config: Config,
+  claims: string[],
+  redirectUri: string
+): void => {
+  const returnCode = config.getReturnCode();
+  if (returnCode) {
+    if (claims.includes("https://vocab.account.gov.uk/v1/returnCode")) {
+      throw new ReturnCodeError(`Return Code: ${returnCode.code}`, redirectUri);
+    } else {
+      throw new ReturnCodeError(
+        "Access Denied: Not requested to view return code",
+        redirectUri
+      );
```

Comment:
these errors feel like two very different things - one is that identity has returned a return code, and the other is the RP has asked for something they're not allowed. I think it's a bit misleading to cover them both by one error?